io/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+#![doc = include_str!("../README.md")]
+#![deny(missing_docs, missing_debug_implementations)]
+#![no_std]
+
+use core::ffi::c_void;
+use mc_sgx_core_sys_types::sgx_status_t;
+use mc_sgx_core_types::Error;
+use mc_sgx_util::ResultInto;
+
+/// Standard Error stream
+#[derive(Debug)]
+pub struct StdError;
+
+impl StdError {
+    /// Attempts to write the entire buffer into the hosts stderr sink.
+    ///
+    /// # Arguments
+    /// * `buffer` - The buffer to write.
+    ///
+    /// # Errors
+    /// If there is any error writing all of `buffer` to the sink. No
+    /// assumptions should be made about the amount that was written to the sink
+    /// when an error occurs.
+    pub fn write_all(&self, buffer: &[u8]) -> Result<(), Error> {
+        unsafe { ocall_stderr(buffer.as_ptr() as *const c_void, buffer.len()) }.into_result()
+    }
+}
+
+extern "C" {
+    /// The ocall to send stderr messages to
+    ///
+    /// # Arguments
+    /// * `input` - The input buffer/stream. Should be ui8/bytes
+    /// * `len` - The byte length of `input`
+    ///
+    /// # Returns
+    /// `sgx_status_t::SGX_SUCCESS` when all of input was successfully written
+    /// to the untrusted stderr sink.
+    /// An error status if not all of the data could be written to the sink. No
+    /// assumptions are made about how much data was written on error.
+    fn ocall_stderr(input: *const c_void, len: usize) -> sgx_status_t;
```

Comment:
Added a task to #17 I'm wondering if this declaration should be in a bindgen generated file which is generated via the edl generated headers